features/environment.py

```diff
@@ -496,31 +470,19 @@ def after_step(context, step):
                 )
 
         if hasattr(context, "machines") and SUT in context.machines:
-            if not os.path.exists(artifacts_dir):
-                os.makedirs(artifacts_dir)
-            for log_file in FAILURE_FILES:
-                artifact_file = os.path.join(
-                    artifacts_dir, os.path.basename(log_file)
-                )
-                logging.info(
-                    "-- pull instance:{} {}".format(log_file, artifact_file)
-                )
-                try:
-                    result = context.machines[SUT].instance.execute(
-                        ["cat", log_file], use_sudo=True
-                    )
-                    content = result.stdout if result.ok else ""
-                except RuntimeError:
-                    content = ""
-                with open(artifact_file, "w") as stream:
-                    stream.write(content)
-            for artifact_file, cmd in FAILURE_CMDS.items():
-                result = context.machines[SUT].instance.execute(
-                    cmd, use_sudo=True
-                )
-                artifact_file = os.path.join(artifacts_dir, artifact_file)
-                with open(artifact_file, "w") as stream:
-                    stream.write(result.stdout)
+            context.machines[SUT].instance.execute(
```

Comment:
Should we have a `try..except` block for the execute commands ? I am worried if we fail here somehow we might not cleanup the machines after (even though we have the cleanup scripts)

Reply:
yep good point I'll add that